sdk/core/Azure.Core/src/Internal/RehydrationOperation.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+    internal class RehydrationOperation : Operation
+    {
+        private readonly NextLinkOperationImplementation _nextLinkOperation;
+        private readonly OperationInternal _operation;
+
+        public RehydrationOperation(HttpPipeline pipeline, RehydrationToken? rehydrationToken, ClientOptions options = null)
+        {
+            AssertNotNull(pipeline, nameof(pipeline));
+            AssertNotNull(rehydrationToken, nameof(rehydrationToken));
```

Comment:
why not use `Argument.AssertNotNull`?

(same in a few other places)

Reply:
updated